### PR TITLE
09 - category filter

### PR DESCRIPTION
**Pull request by stormytuna**

### CODE REVIEW

File: src/components/Reviews.jsx [hunk 1]
```diff
@@ -4,27 +4,47 @@ import { Review } from "./Review";
...
 
   if (isLoading) {
     return <h2>Loading...</h2>;
   }
 
   return (
     <div className="Reviews">
+      <select name="Category" onChange={(e) => setChosenCategory(e.target.value)}>
+        {categories.map((category) => {
+          return <option value={category}>{category}</option>;
```
Comment (Poonam-raj) [inline src/components/Reviews.jsx]: make sure you put a 'key' on the option tag as this is a map (see console warnings)

File: src/components/Reviews.jsx [hunk 2]
```diff
@@ -4,27 +4,47 @@ import { Review } from "./Review";
...
 
   useEffect(() => {
+    let url = "https://be-nc-games.onrender.com/api/reviews";
+    if (chosenCategory !== "all") {
+      url += `?category=${chosenCategory}`;
+    }
+
     axios
-      .get("https://be-nc-games.onrender.com/api/reviews")
+      .get(url)
```
Comment (Poonam-raj) [inline src/components/Reviews.jsx]: Can this be extracted into api.js? Can the axios line on line 23 be a function in api.js that you use instead. Would like all axios requests extracted out please 😁

File: src/components/Reviews.jsx [hunk 3]
```diff
@@ -4,27 +4,47 @@ import { Review } from "./Review";
...
         setReviews(response.data.reviews);
         setIsLoading(false);
+
+        return axios.get("https://be-nc-games.onrender.com/api/categories");
+      })
+      .then((response) => {
+        const fetchedCategories = response.data.categories;
+        const fetchedCategoySlugs = fetchedCategories.map((category) => category.slug);
+        const newCategories = ["all", ...fetchedCategoySlugs];
+        setCategories(newCategories);
```
Comment (Poonam-raj) [inline src/components/Reviews.jsx]: this should be extracted into a component dedicated to listing the categories in a drop down

File: src/components/Reviews.jsx [hunk 4]
```diff
@@ -4,27 +4,47 @@ import { Review } from "./Review";
...
     return <h2>Loading...</h2>;
   }
 
   return (
     <div className="Reviews">
+      <select name="Category" onChange={(e) => setChosenCategory(e.target.value)}>
+        {categories.map((category) => {
+          return <option value={category}>{category}</option>;
+        })}
+      </select>
```
Comment (Poonam-raj) [inline src/components/Reviews.jsx]: This should be extracted into a component dedicated to category drop down, it stops the bloating of this component.

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Heya, I'd like you to extract the axios requests into the api.js file as mentioned previously and extract the drop down functionality into its own component to de-bloat the Reviews component. Let me know any qs as usual 🚀 

thanks!

**Review by Jodiech — APPROVED** [review]

**Pull request merged**